engine.go

```diff
@@ -325,3 +274,66 @@ func (e *EngineBackend) ForkchoiceUpdatedV1(ctx context.Context, heads *types.Fo
 
 	return &types.ForkchoiceUpdatedResult{PayloadStatus: types.PayloadStatusV1{Status: types.ExecutionValid, LatestValidHash: &heads.HeadBlockHash}, PayloadID: &id}, nil
 }
+
+type EthBackend struct {
+	chain *core.BlockChain
+}
+
+func NewEthBackend(chain *core.BlockChain) *EthBackend {
+	return &EthBackend{
+		chain: chain,
+	}
+}
+func (b *EthBackend) Register(srv *rpc.Server) error {
+	srv.RegisterName("eth", b)
+	return node.RegisterApis([]rpc.API{
+		{
+			Namespace:     "eth",
+			Version:       "1.0",
+			Service:       b,
+			Public:        true,
+			Authenticated: false,
+		},
+	}, []string{"eth"}, srv, false)
+}
+
+func (b *EthBackend) rpcMarshalBlock(ctx context.Context, block *ethTypes.Block, inclTx bool, fullTx bool) (map[string]interface{}, error) {
+	fields, err := types.RPCMarshalBlock(block, inclTx, fullTx, b.chain.Config())
+	if err != nil {
+		return nil, err
+	}
+	if inclTx {
```

Comment:
The parameter name is confusing (`inclTx` but really includes `Td` instead), and it's always set to true.

Reply:
I agree. From [geth's ethapi](https://github.com/ethereum/go-ethereum/blob/16701c51697e28986feebd122c6a491e4d9ac0e7/internal/ethapi/api.go#L1200) I'd avoid deviating since it'll be even more confusing!

Reply:
maybe could link to the upstream code in the function docstring 🤔